api/ondemand_vhds_rds_protocol.rst

```diff
@@ -0,0 +1,111 @@
+namespace. The xDS management server will send this same collection of
+resources on reconnect.
+
+.. figure:: diagrams/delta_rds_connection.svg
+   :alt: Delta RDS connection
+
+Requesting Additional resources
+===============================
+In order to request additional resources, Envoy will send a
+:ref:`DeltaDiscoveryRequest <envoy_api_msg_DeltaDiscoveryRequest>`
+to a xDS management server. The xDS management server will then respond with a
+:ref:`DeltaDiscoveryResponse <envoy_api_msg_DeltaDiscoveryResponse>`
+which is followed up by an Ack/Nack
+:ref:`DeltaDiscoveryRequest <envoy_api_msg_DeltaDiscoveryRequest>`
+from Envoy.
+
+.. figure:: diagrams/delta_rds_request_additional_resources.svg
+   :alt: Delta RDS request additional resources
+
+Virtual Host Discovery Service
+==============================
+In order to support on-demand, an additional protocol, VHDS, will be added.
+This protocol allows a separation of concerns with RDS where RDS is in charge
+of maintaining route configs while VHDS is in charge of communicating virtual
+hosts. In VHDS, Envoy will send a
+:ref:`DeltaDiscoveryRequest <envoy_api_msg_DeltaDiscoveryRequest>`
+with :ref:`type_url` set to 
+`type.googleapis.com/envoy.api.v2.route.VirtualHost` 
+and :ref:`resource_names <envoy_api_msg_DeltaDiscoveryRequest.resource_names>` 
+set to a list of routeconfig names + the request HTTP "host"
+headers for which it would like configuration. The management server will
+respond with a
+:ref:`DeltaDiscoveryResponse <envoy_api_msg_DeltaDiscoveryResponse>`
+with the :ref:`resources <envoy_api_msg_DeltaDiscoveryResponse.resources>`
+field populated with the specified virtual hosts, the 
+:ref:`name <envoy_api_msg_DeltaDiscoveryResponse.name>`
+field populated with the virtual host name, and the :ref:`alias` field populated with
+the explicit (no wildcard) domains of the virtual host. Future updates to these
+virtual hosts will be sent via spontaneous updates.
+
+Subscribing to Virtual Hosts
+============================
+Envoy will send a
+:ref:`DeltaDiscoveryRequest <envoy_api_msg_DeltaDiscoveryRequest>`
+with the :ref:`resource_names_subscribe <envoy_api_msg_DeltaDiscoveryRequest.resource_names_subscribe>` field populated with the route config names
++ domains of each of the resources that it would like to subscribe to. Each of
+the virtual hosts contained in the
+:ref:`DeltaDiscoveryRequest's <envoy_api_msg_DeltaDiscoveryRequest>`
+resources field will be added to the route configuration maintained by Envoy.
+If Envoy's route configuration already contains a given virtual host, it will
+be overwritten by data received in the
+:ref:`DeltaDiscoveryResponse <envoy_api_msg_DeltaDiscoveryResponse>`
+but only if the updated virtual host is different from its current state.
+During spontaneous updates configuration server will only send updates for
+virtual hosts that Envoy is aware of. The configuration server needs to
+keep track of virtual hosts known to Envoy.
```

Comment:
Probably worth adding here that if a domain isn't known to the management server, a Resource with an empty "resource" field is sent back. The alias field contains the domain name used in the request.

Reply:
Described in the next paragraph.